testbed/tests/trace_test.go

```diff
@@ -349,16 +405,139 @@ func TestTraceAttributesProcessor(t *testing.T) {
 			// Create a service name that does not match "include" filter.
 			nodeToExclude := &commonpb.Node{ServiceInfo: &commonpb.ServiceInfo{Name: "service-not-to-add-attr"}}
 
-			verifySingleSpan(t, tc, nodeToExclude, spanToInclude, func(span *tracepb.Span) {
+			verifySingleJaegerSpan(t, tc, nodeToExclude, spanToInclude, func(span *tracepb.Span) {
+				// Verify attributes was not added.
+				assert.Nil(t, span.Attributes)
+			})
+
+			// Create another span that does not match "include" filter.
+			spanToExclude := &tracepb.Span{
+				Name: &tracepb.TruncatableString{Value: "span-not-to-add-attr"},
+			}
+			verifySingleJaegerSpan(t, tc, nodeToInclude, spanToExclude, func(span *tracepb.Span) {
 				// Verify attributes was not added.
 				assert.Nil(t, span.Attributes)
 			})
+		})
+	}
+}
+
+// verifySingleZipkinSpan sends a single span to Collector, waits until the span is forwarded
+// and received by MockBackend and calls user-supplied verification function on
+// received span.
+func verifySingleZipkinSpan(
```

Comment:
What is the difference of this function from verifySingleSpan? Can we use the same verification function for Zipkin?